js/bitmart.js

```diff
@@ -712,9 +712,9 @@ module.exports = class bitmart extends Exchange {
         //     }
         //
         const timestamp = this.safeTimestamp2 (ticker, 'timestamp', 's_t', this.milliseconds ());
-        let marketId = this.safeString2 (ticker, 'symbol', 'contract_id');
-        marketId = this.safeString (ticker, 'contract_symbol', marketId);
-        const symbol = this.safeSymbol (marketId, market);
+        const marketId = this.safeString2 (ticker, 'symbol', 'contract_id');
+        market = this.safeMarket (marketId, market, '_');
```

Comment:
In this case, a delimiter would not work for contracts, cause they don't include a delimiter, and in general have a different naming structure for contract ids, so we should remove the 3rd argument (delimiter) from the `safeMarket` call here. We will improve contract id parsing later.